### PR TITLE
Unused parameter analyzer shouldn't warn for methods that only throw

### LINKED ISSUE #65275: IDE0060 shouldn't warn for throw-only methods
IDE0060 currently fires for methods like:
```C#
private static void Blah(int something) => throw new PlatformNotSupportedException();
```
We have many such constructs, due to multitargeting and having the same method in multiple files, in some fully implemented and in others just throwing an exception.  We're having to suppress IDE0060 in many places to account for this.  It'd be really helpful if either always or configurably IDE0060 could support not warning in cases where the method implementation is simply to always throw.

### PR DESCRIPTION
**Pull request by akhera99**
Fixes #65275

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs [hunk 1]
```diff
@@ -1647,5 +1647,28 @@ public static ICustomMarshaler GetInstance(string [|s|])
...
 ");
         }
+
+        [Fact, WorkItem(47174, "https://github.com/dotnet/roslyn/issues/65275")]
+        public async Task TestMethodWithUnusedParameterThrowsExpressionBOdy()
+        {
+            await TestDiagnosticMissingAsync(
+@"public class Class
+{
+    public void Method(int [|x|]) => throw new System.Exception();
```
Comment (ryzngard) [inline src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:1657]: Is there a test that makes sure it still reports if there's more than just a throw?

File: src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs [hunk 2]
```diff
@@ -6,10 +6,12 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Linq;
+using System.Threading;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.FlowAnalysis.SymbolUsageAnalysis;
+using Microsoft.CodeAnalysis.LanguageService;
```
Comment (ryzngard) [inline src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs:14]: needed?

**Review by ryzngard — APPROVED** [review]

File: src/Analyzers/CSharp/Analyzers/RemoveUnusedParametersAndValues/CSharpRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs [hunk 3]
```diff
@@ -33,6 +35,22 @@ protected override bool MethodHasHandlesClause(IMethodSymbol method)
...
+            var methodSyntax = (MethodDeclarationSyntax)node;
+            if (methodSyntax.ExpressionBody is not null)
+            {
+                return methodSyntax.ExpressionBody.Expression is ThrowExpressionSyntax;
+            }
+
+            if (methodSyntax.Body is not null && methodSyntax.Body.Statements.Count == 1)
+            {
+                return methodSyntax.Body.Statements[0] is ThrowStatementSyntax;
+            }
```
Comment (Youssef1313) [inline src/Analyzers/CSharp/Analyzers/RemoveUnusedParametersAndValues/CSharpRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs:49]:
```suggestion
            return methodSyntax.Body is { Statements: [ThrowStatementSyntax] };
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
Can we have a test like:

```csharp
public class C
{
    public C() => throw new System.Exception();
}
```

I'm expecting the cast to fail in C# implementation. It should probably be `BaseMethodDeclarationSyntax`, but even then, I'm not sure if it can fail. Similar thing to VB, though it won't crash due to `TryCast` usage, but could lead to unintended behavior.

It looks like the analyzer currently doesn't support local functions, but it would be good to add a test to make sure things still work properly if local functions are supported in future.

**Pull request merged**